### ISSUE TITLE
Adding HTLC too slow: killing channel

### ISSUE THREAD

**Issue by sha-265** [ElementsProject/lightning#2364]
I'm running c-lightning 0.6.3 on Debian Strech 32bit version, and I made few payments before, but know I'm failing to send payments, even to the node I opened a channel to (1 hop), but I'm getting an error `Adding HTLC too slow: killing channel`, as seen in this log:

> [SKIPPED] 34
2019-02-18T23:06:27.869Z [INFO] lightningd(1345): Server started with public key ..., alias ... (color #0272e5) and lightningd v0.6.3
[SKIPPED] 237
2019-02-18T23:07:32.699Z [INFO] lightningd(1345): pay(0x1b3780c): sendpay via route: us -> 563078:751:0 (42605878msat, 10blk) -> ...
2019-02-18T23:07:32.701Z [INFO] lightningd(1345): Sending 42605878 over 1 hops to deliver 42569086
[SKIPPED] 3
2019-02-18T23:08:02.704Z [UNUSUAL] lightningd(1345): lightning_channeld-... chan #1: Adding HTLC too slow: killing channel
2019-02-18T23:08:02.705Z [INFO] lightningd(1345): ... chan #1: Peer transient failure in CHANNELD_NORMAL: lightning_channeld: Owning subdaemon lightning_channeld died (9)
[SKIPPED] 1
2019-02-18T23:08:02.708Z [INFO] lightningd(1345): pay(0x1b3780c): Try another route
[SKIPPED] 436

### `getinfo` output
```
{
  "id": "...",
  "alias": "...",
  "color": "0272e5",
  "num_peers": 1,
  "num_pending_channels": 0,
  "num_active_channels": 1,
  "num_inactive_channels": 0,
  "address": [
  ],
  "binding": [
    {
      "type": "ipv6",
      "address": "::",
      "port": 9735
    },
    {
      "type": "ipv4",
      "address": "0.0.0.0",
      "port": 9735
    }
  ],
  "version": "v0.6.3",
  "blockheight": 563660,
  "network": "bitcoin",
  "msatoshi_fees_collected": 0
}
```

Any idea why?

**Comment by rustyrussell** [comment]
Hmm, for some reason it's not managing to send the HTLC, even after 30 seconds.  You can use 'log-level=debug' and get *lots* of output; you might prefer to email to me (rusty@blockstream.com) if there's stuff in there you don't want exposed.

If it happens occasionally, I'd expect it's network issues.  But all the time?  That's something else...

**Comment by sha-265** [comment]
@rustyrussell I ran c-lightning with the `log-level=debug` flag, and tried to make a payment. This is what I saw in the log, after trying to make the payment:
>2019-02-19T09:48:35.197Z lightning_gossipd(27714): Trying to find a route from ... to ... for 42457328 msatoshi
2019-02-19T09:48:35.198Z lightning_gossipd(27714): REPLY WIRE_GOSSIP_GETROUTE_REPLY with 0 fds
2019-02-19T09:48:35.198Z lightningd(27702): pay(0xfd8f3c): sendpay via route: us -> 563078:751:0 (42457328msat, 10blk) -> ...
2019-02-19T09:48:35.200Z lightningd(27702): Sending 42457328 over 1 hops to deliver 42374356
2019-02-19T09:48:58.544Z lightningd(27702): ... feerate estimate for slow hit floor 253
2019-02-19T09:49:05.206Z lightningd(27702): lightning_channeld-... chan #1: Adding HTLC too slow: killing channel
2019-02-19T09:49:05.206Z lightningd(27702): lightning_channeld-... chan #1: Status closed, but not exited. Killing
2019-02-19T09:49:05.207Z lightningd(27702): ... chan #1: Peer transient failure in CHANNELD_NORMAL: lightning_channeld: Owning subdaemon lightning_channeld died (9)
2019-02-19T09:49:05.207Z lightningd(27702): ... chan #1: Will try reconnect in 1 seconds
2019-02-19T09:49:05.207Z lightningd(27702): ... chan #1: Failing HTLC 18446744073709551615 due to peer death
2019-02-19T09:49:05.209Z lightningd(27702): Reporting route failure to gossipd: 0x1007 (WIRE_TEMPORARY_CHANNEL_FAILURE) node ... channel 563078:751:0 update
2019-02-19T09:49:05.209Z lightningd(27702): pay(0xfd8f3c): Try another route
2019-02-19T09:49:08.525Z lightning_gossipd(27714): Received routing failure 0x1007 (WIRE_TEMPORARY_CHANNEL_FAILURE), erring node ..., channel 563078:751:0
2019-02-19T09:49:08.525Z lightning_gossipd(27714): Trying to find a route from ... to ... for 42392632 msatoshi
2019-02-19T09:49:08.525Z lightning_gossipd(27714): find_route: No route to ...
2019-02-19T09:49:08.526Z lightning_gossipd(27714): REPLY WIRE_GOSSIP_GETROUTE_REPLY with 0 fds
2019-02-19T09:49:08.688Z lightning_connectd(27713): Connected out for ...
2019-02-19T09:49:08.855Z lightning_hsmd(27712): Client: Received message 1 from client
2019-02-19T09:49:08.856Z lightning_connectd(27713): Connect OUT to ...
2019-02-19T09:49:08.856Z lightning_connectd(27713): peer_out WIRE_INIT
2019-02-19T09:49:09.015Z lightning_connectd(27713): peer_in WIRE_INIT
2019-02-19T09:49:09.016Z lightning_connectd(27713): UPDATE WIRE_CONNECT_PEER_CONNECTED
2019-02-19T09:49:09.016Z lightning_connectd(27713): UPDATE WIRE_CONNECT_PEER_CONNECTED
2019-02-19T09:49:09.016Z lightningd(27702): ... chan #1: Peer has reconnected, state CHANNELD_NORMAL
2019-02-19T09:49:09.018Z lightningd(27702): lightning_channeld-... chan #1: pid 27987, msgfd 24
2019-02-19T09:49:09.018Z lightningd(27702): ... chan #1: Already have funding locked in (and ready to announce)
2019-02-19T09:49:09.021Z lightning_hsmd(27712): Client: Received message 9 from client
2019-02-19T09:49:09.021Z lightning_hsmd(27712): new_client: 1
2019-02-19T09:49:09.135Z lightningd(27702): lightning_channeld-... chan #1: init LOCAL: remote_per_commit = ..., old_remote_per_commit = ... next_idx_local = 117 next_idx_remote = 117 revocations_received = 116 feerates 5055/5055 (range 253-50550)
2019-02-19T09:49:09.136Z lightning_hsmd(27712): Client: Received message 18 from client
2019-02-19T09:49:09.137Z lightning_hsmd(27712): Client: Received message 18 from client
2019-02-19T09:49:09.137Z lightningd(27702): lightning_channeld-... chan #1: peer_out WIRE_CHANNEL_REESTABLISH
2019-02-19T09:49:09.495Z lightningd(27702): lightning_channeld-... chan #1: peer_in WIRE_CHANNEL_REESTABLISH
2019-02-19T09:49:09.495Z lightningd(27702): lightning_channeld-... chan #1: Got reestablish commit=117 revoke=116
2019-02-19T09:49:09.495Z lightning_gossipd(27714): Attempted to local_add_channel a known channel
2019-02-19T09:49:09.495Z lightningd(27702): lightning_channeld-... chan #1: Exchanging announcement signatures.
2019-02-19T09:49:09.501Z lightning_hsmd(27712): Client: Received message 2 from client
2019-02-19T09:49:09.504Z lightningd(27702): lightning_channeld-... chan #1: peer_out WIRE_ANNOUNCEMENT_SIGNATURES
2019-02-19T09:49:09.505Z lightningd(27702): lightning_channeld-... chan #1: Trying commit
2019-02-19T09:49:09.505Z lightningd(27702): lightning_channeld-... chan #1: Can't send commit: nothing to send
2019-02-19T09:49:09.664Z lightningd(27702): lightning_channeld-... chan #1: peer_in WIRE_ANNOUNCEMENT_SIGNATURES
2019-02-19T09:49:10.064Z lightning_gossipd(27714): Received channel_update for 550353:240:0 with old time 1546471384
2019-02-19T09:49:10.065Z lightning_gossipd(27714): Received channel_update for 516634:885:0 with old time 1545588397
2019-02-19T09:49:10.065Z lightning_gossipd(27714): Received channel_update for 551141:3172:1 with old time 1546419734
2019-02-19T09:49:10.066Z lightning_gossipd(27714): Received channel_update for 550334:1291:0 with old time 1545672863
2019-02-19T09:49:10.067Z lightning_gossipd(27714): Received channel_update for 550338:1497:0 with old time 1546193654
...

If you need more info, please let me know, and I will send it by email. 

> If it happens occasionally, I'd expect it's network issues. But all the time? That's something else...

Yes, it's happening since the problem started, so every payment failed since than, not even one successful payment.

Thanks.

**Comment by niftynei** [comment]
Have you tried sending the payment through a different channel, or opening a different channel to send the payment out through? Based on this log, it looks like the channel peer you're trying to send the payment out through isn't responding.

```
2019-02-19T09:49:05.207Z lightningd(27702): ... chan #1: Failing HTLC 18446744073709551615 due to peer death
```

**Comment by sha-265** [comment]
@niftynei yes, I tried with different channel, same issue. I opened a channel with the same node with Anton's Bitcoin Lightning Wallet, and it's working fine.

**Comment by renepickhardt** [comment]
any progress here? Same problem on my side. channel with one of the LNBIG nodes list peer tells me that no HTLCs are offered, connection is true and lightninc-cli pay does not finnish. Not sure if Lightningd in the background still tries to deliver the payment after I hit control+c. I assume it doesn't as `listpeers | grep "htlcs"` shows that I don't have any offered HTLCs anywhere

**Comment by cdecker** [comment]
No, interrupting the `lightning-cli pay` command will _not_ terminate the payment process. It disconnects from the JSON-RPC, and you can monitor the status with `lightning-cli listpays`

**Comment by rustyrussell** [comment]
AI cleanup of old issues.  If wrong, please open a new issue.


Triage decision summary:
User reports HTLC addition timeout causing channel closure on v0.6.3 (Feb 2019). Issue is old, specific to outdated version, and lacks follow-up indicating whether it persists in current releases.
